examples/raw_stream_json_each_row.rs

```diff
@@ -0,0 +1,54 @@
+use std::str::from_utf8;
+
+use serde::Deserialize;
+
+use clickhouse::Client;
+
+/// An example of streaming raw data row-by-row in an arbitrary format.
+/// In this case, it's JSONEachRow. Similarly, it can be used with CSV, TSV, and others;
+/// the only difference will be in how the data is parsed.
+
+#[tokio::main]
+async fn main() -> clickhouse::error::Result<()> {
+    let client = Client::default().with_url("http://localhost:8123");
+
+    let mut raw_cursor_newline = client
+        .query(
+            "
+                SELECT number, hex(randomPrintableASCII(20)) AS hex_str
+                FROM system.numbers
+                LIMIT 10
+                FORMAT JSONEachRow
+            ",
+        )
+        // By default, ClickHouse quotes (U)Int64 in JSON* family formats;
+        // disable it to simplify this example.
+        .with_option("output_format_json_quote_64bit_integers", "0")
+        .fetch_raw()?
+        .newline();
```

Comment:
Maybe like this?

```suggestion
        .query(
            "
                SELECT number, hex(randomPrintableASCII(20)) AS hex_str
                FROM system.numbers
                LIMIT 10
            ",
        )
        // By default, ClickHouse quotes (U)Int64 in JSON* family formats;
        // disable it to simplify this example.
        .with_option("output_format_json_quote_64bit_integers", "0")
        .with_format("JSONEachRow")
        .fetch_raw()?
        .newline();
```

Reply:
This is possible after a few tweaks to the SQLBuilder. See [this commit](https://github.com/ClickHouse/clickhouse-rs/pull/182/commits/cb22c7d48ac3fb3ea6eaf1b39682715a7fe81e8d).

Reply:
Let's leave the **raw** API as-is. The user will quickly find an invalid `FORMAT` with the first query.